### PR TITLE
samples: added upload from/download into memory samples

### PR DESCRIPTION
**Pull request by shaffeeullah**

### CODE REVIEW

File: samples/snippets/storage_upload_from_memory.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        )
+    )
+
+
+# [END storage_file_upload_from_memory]
+
+if __name__ == "__main__":
+    upload_blob_from_memory(
+        bucket_name=sys.argv[1],
+        source_file_name=sys.argv[2],
```
Comment (Breathtender) [inline samples/snippets/storage_upload_from_memory.py:50]: should source_file_name be contents?

File: samples/snippets/storage_download_into_memory.py [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    )
+
+
+# [END storage_download_file]
+
+if __name__ == "__main__":
+    download_blob_into_memory(
+        bucket_name=sys.argv[1],
+        source_blob_name=sys.argv[2],
+        destination_file_name=sys.argv[3],
```
Comment (Breathtender) [inline samples/snippets/storage_download_into_memory.py:55]: Remove destination_file_name ya?

File: samples/snippets/snippets_test.py [hunk 3]
```diff
@@ -209,6 +217,14 @@ def test_download_blob(test_blob):
         assert dest_file.read()
 
 
+def test_download_blob_into_memory(test_blob):
+    contents = storage_download_into_memory.download_blob_into_memory(
+        test_blob.bucket.name, test_blob.name
+    )
+
+    assert contents == "Hello, is it me you're looking for?"
```
Comment (Breathtender) [inline samples/snippets/snippets_test.py:225]: Very good. Got my code review soundtrack playing now
https://www.youtube.com/watch?v=UBYnT8JY7sE

**Review by unforced — APPROVED** [review]
All good besides those few things pointed to.

File: samples/snippets/snippets_test.py (hunk 3 above)
Comment (Breathtender) [reply samples/snippets/snippets_test.py:225]: Oh, this test isn't passing.
your method doesn't return a value, it prints it out, and so you need to capture the `out`

Like so:

```
    out, _ = capsys.readouterr()
    assert "Hello world" in out
    ```

Comment (Breathtender) [reply samples/snippets/snippets_test.py:225]: For reference:
https://github.com/googleapis/python-storage/pull/645/files

File: samples/snippets/storage_download_into_memory.py [hunk 4]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env python
+
+# Copyright 2019 Google Inc. All Rights Reserved.
```
Comment (cojenco) [inline samples/snippets/storage_download_into_memory.py:3]: nit: update to Copyright 2021

File: samples/snippets/snippets_test.py (hunk 3 above)
Comment (shaffeeullah) [reply samples/snippets/snippets_test.py]: Ah thank you!

File: samples/snippets/storage_download_into_memory.py (hunk 2 above)
Comment (shaffeeullah) [reply samples/snippets/storage_download_into_memory.py]: Good catch! I forgot to change these parts of the file.

File: samples/snippets/storage_upload_from_memory.py (hunk 1 above)
Comment (shaffeeullah) [reply samples/snippets/storage_upload_from_memory.py]: Yep yep. Thank you!!

File: samples/snippets/storage_download_into_memory.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+# limitations under the License.
+
+import sys
+
+# [START storage_download_file]
+from google.cloud import storage
+
+
+def download_blob_into_memory(bucket_name, source_blob_name):
+    """Downloads a blob from the bucket."""
```
Comment (cojenco) [inline samples/snippets/storage_download_into_memory.py:24]: Thanks for adding this! Could we update the description to """Downloads a blob into memory.""" and change the parameter name to `blob_name`? The word source might be a bit confusing?

Comment (shaffeeullah) [reply samples/snippets/storage_download_into_memory.py]: done!

File: samples/snippets/storage_download_into_memory.py (hunk 4 above)
Comment (cojenco) [reply samples/snippets/storage_download_into_memory.py]: Same for the upload sample, thanks! Whoa I'm not sure if it's Google Inc or Google LLC. Let me check.

File: samples/snippets/snippets_test.py [hunk 6]
```diff
@@ -189,6 +191,12 @@ def test_upload_blob(test_bucket):
         )
 
 
+def test_upload_blob_from_memory(test_bucket):
+    storage_upload_from_memory.upload_blob_from_memory(
+        test_bucket.name, "Hello, is it me you're looking for?", "test_upload_blob"
+    )
+
```
Comment (cojenco) [inline samples/snippets/snippets_test.py:198]: You could either assert that the blob now exists or assert capsys.readouterr() here just to verify that the print statement successfully ran.

**Review by cojenco — COMMENTED** [review]
Thanks @shaffeeullah for adding the samples🎉 Just a few comments. Also could you help update the samples readme ([ref PR](https://github.com/googleapis/python-storage/pull/656/files)), thanks!

File: samples/snippets/storage_download_into_memory.py (hunk 4 above)
Comment (shaffeeullah) [reply samples/snippets/storage_download_into_memory.py]: it seems the 2021 ones have Google LLC. updated

Comment (cojenco) [reply samples/snippets/storage_download_into_memory.py]: yep thanks, i just checked OSPO, and it's Google LLC

**Review by cojenco — APPROVED** [review]

**Pull request merged**